sql/parser/eval.go

```diff
@@ -567,13 +568,14 @@ var defaultContext = EvalContext{
 }
 
 // makeDDate constructs a DDate from a time.Time in the session time zone.
-func (ctx EvalContext) makeDDate(t time.Time) (Datum, error) {
+func (ctx EvalContext) makeDDate(t time.Time) (DDate, error) {
 	loc, err := ctx.GetLocation()
 	if err != nil {
-		return DNull, err
+		return DummyDate, err
 	}
 	year, month, day := t.In(loc).Date()
-	return DDate{Time: time.Date(year, month, day, 0, 0, 0, 0, time.UTC)}, nil
+	secs := time.Date(year, month, day, 0, 0, 0, 0, time.UTC).Unix()
+	return DDate(secs / driver.SecondsInDay), nil
```

Comment:
I think not repeating yourself by using the `driver.SecondsInDay` constant is not worth the dependency on the `sql/driver` package. I'd probably feel differently if the constant could change, but `SecondsInDay` is a constant in the true sense: it can never have a value other than `24 * 60 * 60`.